src/server/storage/storageAPI.js

```diff
@@ -179,5 +181,54 @@ const setup = (app: Router, storage: StorageAPI) => {
       res.json({ ok: 1, result })
     })
   )
+
+  /**
+   * @api {get} /login/token get W3 login token for current user
+   * @apiName Get
+   * @apiGroup Storage
+   *
+   * @apiSuccess {Number} ok
+   * @apiSuccess {String} loginToken
+   * @ignore
+   */
+  app.get(
+    '/storage/login/token',
+    passport.authenticate('jwt', { session: false }),
+    wrapAsync(async (req, res, next) => {
+      const { user, log } = req
+      const logger = log.child({ from: 'storageAPI - login/token' })
+
+      let loginToken = user.loginToken
+
+      if (!loginToken) {
+        const secureHash = md5(user.email + conf.secure_key)
+        const web3Response = await fetch(`${conf.web3SiteUrl}/api/wl/user`, {
+          method: 'PUT',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            secure_hash: secureHash.toLowerCase(),
+            email: user.email
+          })
+        })
+          .then(res => res.json())
+          .catch(e => {
+            logger.error('Get Web3 Login Response Failed', e)
```

Comment:
What will be in the answer if the request falls?  loginToken = ?
You always return 
`res.json({
        ok: +Boolean(loginToken),
        loginToken
      })`

Reply:
@StanislavShevchenko that is ok. If an error will occur during request to w3 api the GoodServer will return { ok: 0 } in response body to the GoodDapp. The GoodDapp will see that there is no login_token and handle it.